cognite_toolkit/_cdf_tk/commands/_base.py

```diff
@@ -29,29 +28,18 @@ class ToolkitCommand:
     def __init__(self, print_warning: bool = True, skip_tracking: bool = False, silent: bool = False):
         self._print_warning = print_warning
         self.silent = silent
-        if len(sys.argv) > 1:
-            self.user_command = f"cdf-tk {' '.join(sys.argv[1:])}"
-        else:
-            self.user_command = "cdf-tk"
         self.warning_list = WarningList[ToolkitWarning]()
-        self.tracker = Tracker(self.user_command)
-        self.skip_tracking = self.tracker.opted_out or skip_tracking
+        self.tracker = Tracker(skip_tracking)
 
     @property
     def print_warning(self) -> bool:
         return self._print_warning and not self.silent
 
     def _track_command(self, result: str | Exception) -> None:
-        if self.skip_tracking or "PYTEST_CURRENT_TEST" in os.environ:
-            return
-        self.tracker.track_command(self.warning_list, result, type(self).__name__.removesuffix("Command"))
+        self.tracker.track_cli_command(self.warning_list, result, type(self).__name__.removesuffix("Command"))
 
     def run(self, execute: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
-        if (
-            not self.tracker.opted_in
-            and not self.tracker.opted_out
-            and not self.user_command.startswith("cdf-tk collect")
-        ):
+        if not self.tracker.opted_in and not self.tracker.opted_out and "collect" not in sys.argv:
```

Comment:
Maybe a bit risky? In case we have collect in some flag or subcommand. What about?
```suggestion
        if not self.tracker.opted_in and not self.tracker.opted_out and (len(sys.argv) > 2"collect" == sys.argv[1]):
```
Ensure you verify that `collect will be the second argument.